sql-plugin/src/main/scala/com/nvidia/spark/rapids/Arm.scala

```diff
@@ -156,6 +158,34 @@ trait Arm {
       h.close()
     }
   }
+
+  def withRetry(
+      input: SpillableColumnarBatch)
+      (fn: ColumnarBatch => ColumnarBatch): Seq[SpillableColumnarBatch] = {
+    val retry = new RmmRapidsRetryHelper(input, null, null)
```

Comment:
Nit: Not a fan of passing null arguments to a constructor/method.  Should there be an alternate constructor of RmmRapidsRetryHelper that just takes an input?

Reply:
added multiple constructors here.